### PR TITLE
docker_push for docker_bundle

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
@mikedanese FYI this should implement the `docker_bundle` version of `docker_push`.  It is sitting on top of my larger refactoring changes, I'll ping back here when this is suitable for review in isolation, but if you want to kick the tires it should work.

**Comment by mattmoor** [comment]
The diffbase is in, so this is RFAL.  @dlorenc want to do the honors?

File: docker/contrib/push-all.bzl [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+def docker_push(image=None, **kwargs):
+  """Pushes the docker images from a bundle.
+
+  This rule pushes N docker images to a Docker registry.
+
+  Args:
+    name: name of the rule
+    bundle: the docker_bundle to publish.
+  """
+  _docker_push(image=image, **kwargs)
```
Comment (damienmg) [inline docker/contrib/push-all.bzl]: I don't see the image attribute on _docker_push

File: docker/contrib/push-all.sh.tpl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xe
```
Comment (damienmg) [inline docker/contrib/push-all.sh.tpl]: Why have a template and not use ctx.file directly?

**Comment by mattmoor** [comment]
This is RFAL

File: docker/contrib/push-all.bzl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            cfg = "host",
+            executable = True,
+            allow_files = True,
+        ),
+    },
+    executable = True,
+    implementation = _impl,
+)
+
+def docker_push(bundle=None, **kwargs):
```
Comment (damienmg) [inline docker/contrib/push-all.bzl]: Why have a macro wrapper? Why not make the rule directly named docker_push?

**Comment by mattmoor** [comment]
Done.

**Comment by ixdy** [comment]
probably should update docs, too?

**Comment by mattmoor** [comment]
@ixdy right now the stuff under `contrib` is lacking documentation altogether.  IIRC we'd stopped short of adding it because we anticipated a quicker [switch to skydoc](https://github.com/bazelbuild/rules_docker/issues/32) than has happened.  

@davidzchen Any update on the blocking issue?

**Pull request merged**